scene/main/node.cpp

```diff
@@ -3037,6 +3037,19 @@ PackedStringArray Node::get_configuration_warnings() const {
 	ERR_THREAD_GUARD_V(PackedStringArray());
 	PackedStringArray ret;
 
+#ifdef TOOLS_ENABLED
```

Comment:
`get_configuration_warnings()` is de facto an editor only API, so the guards aren't needed here, at least we don't add them usually in other classes.

Reply:
I wonder if this means that this code is compiled in export templates, which would increase their binary size. That said, there's been discussion about being able to print configuration warnings in debug export templates at least, so if we add guards to them, it'd probably be `#ifdef DEBUG_ENABLED` instead.

Reply:
Yeah we should test that. Maybe all `get_configuration_warnings()` implementations would need to be enclosed in `TOOLS_ENABLED`.

Reply:
Just like `get_argument_options` was [discussed last time](https://github.com/godotengine/godot/pull/66427#pullrequestreview-1141152103).